Automattic/wp-calypso
# Framework: CSS: Add stylizer component

This PR is the next part of https://github.com/Automattic/wp-calypso/issues/11959, continuing the work done in https://github.com/Automattic/wp-calypso/issues/12451. It adds a new higher order component, called `Stylizer`. This outputs the CSS for a component when the component is mounted, so that we only load the CSS when it is needed.

I have implemented this for one component - `VerticalNav` as a demonstration of how it will work, and that it does work.

We use Webpack to load all required dependencies and wrap with Stylizer. There are the following items which need to be resolved in the follow up PRs before we start to roll out it on a wider scale:

* We should import style.scss file only when it's present in the folder where React component is located
* RTL support is missing
* SSR support
* Documentation
  
#### Testing instructions
  
1. Run `git checkout add/stylizer` and start your server, or open a [live branch](https://delphin.live/?branch=add/stylizer)
2. Open the [`Domain management` page](http://calypso.localhost:3000/domains/manage/example.com/edit/example.com)
3. Check that the vertical nav CSS is loaded in the head:
<img width="617" alt="screen shot 2017-04-27 at 10 15 37" src="https://cloud.githubusercontent.com/assets/275961/25476767/2483ee4a-2b33-11e7-98f7-bfe435c7eb7f.png">
4. Navigate to the reader, and check that the CSS is removed from the head.

#### Reviews
  
   
@Automattic/sdev-feed
